File: Pregunta3/Python/OrdenSuperior.py
def OrdenSuperior(n, fun):
	return fun(n)
def fibo(n):
    if n == 0: return 0
    if n == 1: return 1
    if n == 2: return 1
    return fibo(n-1) + fibo(n-2) + fibo(n-3)

print ("\nFibonacci Funcion Orden Superior")
n = int(input("Ingrese un N : "))
for i in range(n):
    print(OrdenSuperior(i, fibo), end=", ")

File: Pregunta3/Haskell/OrdenSuperior.hs
fibo :: Int -> Int
fibo 1 = 0
fibo 2 = 1
fibo 3 = 1
fibo n | n > 3 = fibo(n-1) + fibo(n-2) + fibo(n-3)

ordenSuperior :: Int -> [Int]
ordeSuperior n = [fibo(x) | x <- [1..n], x > 0]


main = do
 putStrLn "\nFibonacci Funcion Orden Superior"
 putStrLn "Introduzca un N : "
 n<-getLine
 putStrLn "Los N primeros fibonacci "
 print(ordenSuperior(read n :: Int))
